pkg/ui/workspaces/cluster-ui/src/store/databaseDetails/databaseDetails.reducer.ts

```diff
@@ -30,15 +37,26 @@ export type DatabaseDetailsState = {
   inFlight: boolean;
 };
 
+export type DatabaseDetailsSpanStatsState = {
```

Comment:
WDYT about using the `RequestState` type here and also when creating the details object you can pass the entire request state as a single object instead of spreading them into fields lie `loaded, loading, spanStatsError` etc.